src/PIL/EpsImagePlugin.py

```diff
@@ -333,22 +341,38 @@ def _read_comment(s: str) -> bool:
                 else:
                     break
 
-                self._size = columns, rows
-                return
+                # Parse the columns and rows after checking the bit depth and mode
+                # in case the bit depth and/or mode are invalid.
+                imagedata_size = columns, rows
             elif bytes_mv[:5] == b"%%EOF":
                 break
             elif trailer_reached and reading_trailer_comments:
                 # Load EPS trailer
                 s = str(bytes_mv[:bytes_read], "latin-1")
-                _read_comment(s)
+                read_comment(s)
             elif bytes_mv[:9] == b"%%Trailer":
                 trailer_reached = True
             bytes_read = 0
 
-        if not self.tile:
+        # A "BoundingBox" is always required,
+        # even if an "ImageData" descriptor size exists.
+        if not bounding_box:
             msg = "cannot determine EPS bounding box"
             raise OSError(msg)
 
+        # An "ImageData" size takes precedence over the "BoundingBox".
+        if imagedata_size:
+            self._size = imagedata_size
+        else:
+            self._size = (
+                bounding_box[2] - bounding_box[0],
+                bounding_box[3] - bounding_box[1],
+            )
+
+        self.tile = [
+            ImageFile._Tile("eps", (0, 0) + self._size, offset, (length, bounding_box))
```

Comment:
```suggestion
            ImageFile._Tile("eps", (0, 0) + self.size, offset, (length, bounding_box))
```
I'd like to save `self._size` for when it is being assigned to.

Reply:
huh, I don't remember changing that. I'm not sure how that happened.